### PR TITLE
MAILBOX-403 Email main body is also indexed as an attachment

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java [hunk 1]
```diff
@@ -303,7 +303,8 @@ private boolean isHtmlSubType(MimePart mimePart) {
     @JsonIgnore
     public Stream<MimePart> getAttachmentsStream() {
         return attachments.stream()
-                .flatMap((mimePart) -> Stream.concat(Stream.of(mimePart), mimePart.getAttachmentsStream()));
+                .flatMap(mimePart -> Stream.concat(Stream.of(mimePart), mimePart.getAttachmentsStream()))
+                .filter(mimePart -> mimePart.contentDisposition.isPresent());
```
Comment (Arsnael) [inline mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java:307]: Not sure to understand fully this, can you explain?

File: mailbox/store/src/test/resources/eml/recursiveMail.json [hunk 2]
```diff
@@ -77,38 +77,6 @@
...
-      "contentDisposition": null
-    },
-    {
-      "textContent": "Textual content of one attachment ( mail signature )\n",
-      "mediaType": "text",
-      "subtype": "plain",
-      "fileName": null,
-      "fileExtension": null,
-      "contentDisposition": null
-    },
```
Comment (Arsnael) [inline mailbox/store/src/test/resources/eml/recursiveMail.json:111]: Why do we need to remove attachments in tests?

Comment (chibenwa) [reply mailbox/store/src/test/resources/eml/recursiveMail.json:111]: Content disposition is null, it is not an attachment

File: mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java:307]: So far we considered all mime parts as attachments, even 'main' body parts, and multiparts.

That's not what an attachment is.

An attachment is a mimepart, but it needs a `Content-Disposition` header, typically with a value `inline` or `attachment`. Without it, it is just NOT a mime part, indexing that mime part is both functionnally and performance wise plenty wrong.

Comment (Arsnael) [reply mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/MimePart.java:307]: Thanks for the explanation

**Review by Arsnael — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

File: mailbox/store/src/test/resources/eml/recursiveMail.json [hunk 3]
```diff
@@ -77,38 +77,6 @@
   ],
   "sentDate": "2015-06-10T10:45:27+02:00",
   "attachments": [
-    {
-      "textContent": "Forward as attachment !\n\n\n",
```
Comment (rouazana) [inline mailbox/store/src/test/resources/eml/recursiveMail.json:81]: I'm surprised by this change. If you forward an email as an attachment, it is an attachment, no?

Comment (chibenwa) [reply mailbox/store/src/test/resources/eml/recursiveMail.json:81]:
```
--------------080603090509090707040003
Content-Type: text/plain; charset=utf-8; format=flowed
Content-Transfer-Encoding: 7bit

Forward as attachment !
```


It's the main body! Not an embeded email. Looks irrelevant to me!

Comment (rouazana) [reply mailbox/store/src/test/resources/eml/recursiveMail.json:81]: yes sorry

**Pull request closed without merging**